resources/api/resource.py

```diff
@@ -317,9 +320,49 @@ def filter_is_favorite(self, queryset, name, value):
         else:
             return queryset.exclude(favorited_by=self.user)
 
+    def _deserialize_datetime(self, value):
+        try:
+            return arrow.get(value).datetime
+        except ParserError:
+            raise exceptions.ParseError("'%s' must be a timestamp in ISO 8601 format" % value)
+
+    def filter_available_between(self, queryset, name, value):
+        if len(value) != 2:
+            raise exceptions.ParseError('available_between takes exactly two comma-separated values.')
+
+        available_start = self._deserialize_datetime(value[0])
+        available_end = self._deserialize_datetime(value[1])
+
+        if available_start.day != available_end.day:
+            raise exceptions.ParseError('available_between timestamps must be on the same day.')
+
+        reservations = Reservation.objects.filter(end__gt=available_start).filter(begin__lt=available_end)
+        queryset = queryset.exclude(reservations__in=reservations)
+
+        not_available_ids = []
+        for resource in queryset:
+            opening_hours = resource.get_opening_hours(available_start, available_end)
```

Comment:
FWIW, I think it'd make for prettier code if this loop body was refactored into a function of its own that would just return `True`/`False` depending on availability, instead of having to use `continue`s.

Then `not_available_ids` could be computed simply by way of 

```python
not_available_ids = set(
    resource.id
    for resource in resources
    if not _is_available_for_blah_whatever(resource, available_start, available_end)
)
```
and the function would also be more easily testable